### PR TITLE
Changed all is_*? method names to *?

### LINKED ISSUE #264: `is_empty?` should be replaced with `empty?` (Or at least alias it for a smoother transition). This goes for every `is_*?` method.

### PR DESCRIPTION
**Pull request by stephenandersondev**
### Summary

Changed all is_*? method names to *? on the following:

- is_empty? to empty?
- is_simple? to simple?
- is_closed? to closed?
- is_point? to point?
- is_ring? to ring?
- is_capi_geos? to capi_geos?
- is_ffi_geos? to ffi_geos?
- is_geos? to goes?

Closes #264

### CODE REVIEW

**Comment by stephenandersondev** [comment]
This is my first open-source contribution, so please let me know if I've missed any steps.

**Comment by keithdoggett** [comment]
@stephenandersondev thanks for submitting this I think this looks good, but I'll take a deeper look soon. @BuonOmo do you think we should alias all the methods to accept the is_*? form as well? It would be good for backwards compatibility.

Also @stephenandersondev this is going to be part of our V3 release, so this needs to be merged into the `v3-dev` branch.

**Comment by stephenandersondev** [comment]
@keithdoggett Just changed the branch to `v3-dev`, thanks for the heads up!

**Comment by BuonOmo** [comment]
> This is my first open-source contribution ever, so please let me know if I've missed any steps. Thanks!

Thank you for doing it there!

---

@keithdoggett @stephenandersondev I thought a bit about it, and I think we can do some easier transition for users, that would even allow us to merge this in v2.

I'm thinking about keep `is_*` methods as well for a transition period, yet printing a deprecation warning. For instance:

```ruby
def is_simple?
	warn "is_*? methods are deprecated, please use the *? counterpart, will be removed in v3" unless ENV["RGEO_SILENCE_DEPRECATION"]

	simple?
end
```

Of course, this might also be done using metaprogramming (for instance with `method_missing`), yet I'm not sure this is necessary.


What do you think about that idea, and @stephenandersondev could you implement it in this PR? 🙂

**Comment by stephenandersondev** [comment]
@BuonOmo Yes I can definitely implement this PR, I will have it ready shortly 👍

**Comment by stephenandersondev** [comment]
@BuonOmo I got the alias methods put in, have a couple of questions though:

- I didn't see a branch that referenced v2 so I assumed it was `master` since it's the current version? Let me know if otherwise and I can change it.
- I wasn't sure if we needed to add the alias methods to the tests? If so, let me know what you think that should look like and I'll get it in.

File: ext/geos_c_impl/geometry.c
Comment (BuonOmo) [inline ext/geos_c_impl/geometry.c:1099]: The geos methods do not implement `Feature` base methods in v2 unfortunately.

Hence we also need to create the deprecation warning. You can do that one in ruby, in `lib/rgeo/geos/capi_feature_classes.rb` :)

File: ext/geos_c_impl/geometry_collection.c
Comment (BuonOmo) [inline ext/geos_c_impl/geometry_collection.c:626]: Same here unfortunately

File: ext/geos_c_impl/line_string.c
Comment (BuonOmo) [inline ext/geos_c_impl/line_string.c:667]: and there

**Review by BuonOmo — COMMENTED** [review]
@stephenandersondev thank you for the addition.

> I didn't see a branch that referenced v2 so I assumed it was master since it's the current version? Let me know if otherwise and I can change it.

You're right, v2 is the current version, we're using the master branch.

> I wasn't sure if we needed to add the alias methods to the tests? If so, let me know what you think that should look like and I'll get it in.

IMHo, that is not needed.

---

I've one last review concerning the geos capi, unfortunately we need alias for those methods as well. I've adressed which methods and where they should be aliased. I can do it and push to your branch later if you don't have more time 🙂 (I'll let you decide though, your branch, your ownership!)

After that I think we'll be ready to ship.

**Comment by stephenandersondev** [comment]
So sorry about that @BuonOmo. I'm admittedly very new to RGeo and this is my first time ever looking at C code 😅  I think I got the aliasing right? Although, you may mean to alias do the aliasing on the ruby side which is where I would add the warning back in at? If this is wrong (which I feel it is) would you be able to do one as an example?  I would be happy to do the rest 🙂  Sorry again, still learning this stuff.

**Comment by BuonOmo** [comment]
@stephenandersondev no problem, I'll give you some pointers.

Let's try with the `empty?` GEOS method for instance. The idea will be to implement `is_empty?` on the ruby side, to have something simpler to code. Performances will not be an issue anyway for such a trivial task (proxying to the actual `empty?` method).

We first need to know where the method is implemented. The line below gives us this information:

```c
rb_define_method(geos_geometry_methods, "empty?", method_geometry_is_empty, 0);
```

<sup>
[`rb_define_method` reference](https://github.com/ruby/ruby/blob/HEAD/doc/extension.rdoc#method-and-singleton-method-definition-)
</sup>

We can see it is defined under `geos_geometry_methods`. And this correspond to the ruby module `XX::CAPIGeometryMethods` according to:

```c
geos_geometry_methods = rb_define_module_under(globals->geos_module, "CAPIGeometryMethods");
```

This module can be found when searching the project, and you'll most likely end up in `lib/rgeo/geos/capi_feature_classes.rb`, you now know you can define the method here:

```ruby
module RGeo
  module Geos
    module CAPIGeometryMethods # :nodoc:
      include Feature::Instance

      def is_empty?
        warn "support..."
        empty?
      end
...
```

Is that clear enough? If not, I'll just takeover this last bit and you'll see my commit as an example 🙂

**Comment by stephenandersondev** [comment]
@BuonOmo Thank you so much for taking the time to clarify this for me, I learned something new because of it 😄  I believe this last commit I pushed does the trick. Please let me know if otherwise.  Thanks again for your help!

**Review by BuonOmo — APPROVED** [review]
LGTM!

**Comment by BuonOmo** [comment]
@keithdoggett do you want to process to one last review?

**Comment by BuonOmo** [comment]
Oh, my bad, rubocop is not passing anymore, @stephenandersondev the trick is to disable the `Naming/PredicateName` cop (see https://docs.rubocop.org/rubocop/1.3/configuration.html#disabling-cops-within-source-code)

**Comment by stephenandersondev** [comment]
So sorry about that 🤦🏻  Just committed the disabling of it. Thank you so much for your patience and guidance on this!

**Comment by BuonOmo** [comment]
@stephenandersondev sorry I should have been a bit clearer.... I think that this should be inlined:

```ruby
def is_simple? # rubocop:disable Naming/PredicateName
```

This will ensure that when getting rid of every `is_` method, we'll also get rid of that comment. Moreover, it tells a ready why this comment is here.

**Comment by stephenandersondev** [comment]
@BuonOmo Oops sorry about that, they're inline now 👍🏻

File: History.md
Comment (keithdoggett) [inline History.md:164]: Not sure we should change this since it's describing a past change.

File: History.md
Comment (keithdoggett) [inline History.md:695]: Not sure we should change this since it's describing a past change.

**Review by keithdoggett — COMMENTED** [review]
Thanks for the work @stephenandersondev and @BuonOmo! These look good for the most part. My only suggestion is that we don't modify the `History.md` file because that's supposed to reflect historical changes that happened and probably shouldn't be modified only appended to.

**Comment by stephenandersondev** [comment]
@keithdoggett Ahhh that's a great point. I just reverted them back. Thanks for the head up!

**Review by keithdoggett — APPROVED** [review]
Awesome, thanks @stephenandersondev! LGTM

@BuonOmo should be good to merge as is unless you want to squash the commits first.

**Comment by BuonOmo** [comment]
Merged! Thanks again @stephenandersondev  :)

**Comment by stephenandersondev** [comment]
Thank you both! I look forward to working with you again :)

**Comment by BuonOmo** [comment]
well if you see something on which you could work on the [roadmap](https://github.com/rgeo/rgeo/issues/247), go ahead 🙂

**Pull request merged**
